src/Client.ts

```diff
@@ -11,18 +12,82 @@ import discord, {
 import { dispatch } from './dispatch';
 import { loadCommands } from './loadCommands';
 import { Command } from './Command';
+import { commandEquals, toData } from './utils/command';
 
 export default class Client extends discord.Client {
+
+  private commands = new Collection<string, Command>();
+
   /**
    * Handles commands for the bot.
    */
   constructor(options: ClientOptions) {
     super(options);
+  }  
+
+  async syncCommands(commands: Command[]): Promise<void> {
+
+    if (!this.isReady()) {
+      throw new Error('This must be used after the client is ready.');
+    }
+
+    if (!process.env.GUILD_ID) {
+      throw new Error('Please set the GUILD_ID in your .env file.');
+    }
+
+    // Fetch the commands from the server.
+    const rawCommands = process.env.NODE_ENV === 'development' ?  
+      await this.guilds.cache.get(process.env.GUILD_ID)?.commands.fetch() : 
+      await this.application.commands.fetch();
+
+    if (!rawCommands) {
+      throw new Error('Could not fetch remote commands!');
+    }
+
+    // Normalize all of the commands.
+
+    const appCommands = new Collection<string, ApplicationCommandData>();
+    rawCommands.map(toData).forEach(data => appCommands.set(data.name, data));
+
+    const clientCommands = new Collection<string, ApplicationCommandData>();
+    commands.map(toData).forEach(data => clientCommands.set(data.name, data));
+
+    // Helper for whenever there's a diff.
+    const push = async () => {
+      console.log('Local commands differ from remote commands, syncing now...');
+      await this.pushCommands(commands);
+      console.log('Finished syncing');
+    };
+
+    // If the length is not the same it's obvious that the commands aren't the same.
+    if (appCommands.size !== clientCommands.size) {
+      console.log({ appCommands: appCommands.size, commands: clientCommands.size });
+      await push();
+      return;
+    }
+
+
+    // Calculate if theres a diff between the local and remote commands.
+    const diff = !appCommands.every(appCommand => {
+      // Get the name, and get the corresponding command with the same name.
+      const { name } = appCommand;
+      const clientCommand = clientCommands.get(name);
```

Comment:
I think you should just replace these lines with `const clientCommand = clientCommands.get(appCommand.name);`